content/chapter-destinations/destination-s3/_index.md

```diff
@@ -0,0 +1,205 @@
+
+Available in {{% param "product.abbrev" %}} version 4.4 and later.
+
+The `s3()` destination sends log messages to the [Amazon Simple Storage Service (Amazon S3)](https://aws.amazon.com/s3/) object storage service. You can send log messages over TCP, or encrypted with TLS.
+
+<!-- FIXME minimal config example, required options -->
+
+<!-- FIXME what logic do we use to create objects in the bucket? -->
+
+## Prerequisites
+
+- An existing S3 bucket configured for programmatic access, and the related `ACCESS_KEY` and `SECRET_KEY` of a user that can access it.
+<!-- FIXME What else needs to be configured from the Amazon side? -->
+
+<!-- 
+required Python dependencies (`boto3` and/or `botocore`)
+    > When does the user have to install these manually?
+-->
+
+To use the `s3()` driver, the `scl.conf` file must be included in your {{% param "product.abbrev" %}} configuration:
+
+```shell
+   @include "scl.conf"
+```
+
+The `s3()` driver is actually a reusable configuration snippet. For details on using or writing such configuration snippets, see {{% xref "/chapter-configuration-file/large-configs/config-blocks/_index.md" %}}. You can find the source of this configuration snippet on [GitHub](https://github.com/syslog-ng/syslog-ng/blob/master/modules/python-modules/syslogng/modules/s3/s3_destination.py).
+
+## Options
+
+The following options are specific to the `s3()` destination.
+
+<!-- FIXME are there any common/generic/inherited options that we should reference? -->
+
+## access-key()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The `ACCESS_KEY` of the service account used to access the S3 bucket. (Together with [`secret-key()`](#secret-key).)
+
+## bucket()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The name of the S3 bucket, for example, `my-bucket`
+
+## canned-acl()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `empty` |
+
+*Description:* The ACL assigned to the object, if specified, for example, `bucket-owner-read`. The following values are valid:
+
+`authenticated-read`, `aws-exec-read`, `bucket-owner-full-control`, `bucket-owner-read`, `log-delivery-write`, `private`, `public-read`, `public-read-write`
+
+If you configure an invalid value, the default is used.
+
+## chunk-size()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `5MiB` |
+
+*Description:* The size of log messages that  {{% param "product_name" %}} writes to the S3 object in a batch.
+<!-- FIXME how does is relate to compression? -->
+
+## compression()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | boolean |
+| Default: | `no` |
+
+*Description:* <!-- FIXME -->
```

Comment:
Setting compression(yes) enables gzip compression, and implicitly adds a .gz suffix to the
created object's key. Use the compresslevel() options to set the level of compression (0-9).